Documentation/architecture.md

```diff
@@ -0,0 +1,58 @@
+# Architecture
+
+## Application setup
+
+Here is the diagram which explains the general workflow of the user with the application setup. On the left you can see that the Adm has created the whole setup using terraform scripts. And once the setup is created the SR can access it via webbrowser on the generated URLs provided by the Adm.
+
+![Architecture Diagram](arch-diagram.png "Architecture Diagram")
+
+The terminal on the browser is backed by a persistant websocket connection.
+
+## Terraform setup
+
+The terraform code in [terraform directory](../terraform) at the root of this project sets up the application on AWS in two VMs.
+
+There are two different VMs currently because one support Apparmor and other supports SELinux. Having both in the same machine is problematic and introduces it's own set of issues so to separate concerns we have put it in two different VMs.
+
+### Code setup
+
+The code tree looks like following:
+
+```bash
+├── flagfiles.tf
+├── locals.tf
+├── machine.tf
+├── output.tf
+├── password.tf
+├── provider.tf
+├── provisioning
+│   ├── common.bash
+│   ├── fedora.bash
+│   └── ubuntu.bash
+├── provision.tf
+├── terraform.tfvars.tmpl
+├── variables.tf
+└── vpc.tf
+```
+
+* `provider.tf` currently the only provider is AWS.
```

Comment:
`provider.tf` configures the provider which currently is only AWS.

Reply:
done!